src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -11476,14 +11527,19 @@ private void SetValue(int id, int index, NullableFlowState value)
                 }
                 else
                 {
-                    // No states should be modified in unreachable code, as there is only one unreachable state.
-                    if (!this.Reachable) return;
-                    index *= 2;
-                    _state[index] = (value != NullableFlowState.NotNull);
-                    _state[index + 1] = (value == NullableFlowState.MaybeDefault);
+                    SetValue(index, value);
                 }
             }
 
+            private void SetValue(int index, NullableFlowState value)
+            {
+                // No states should be modified in unreachable code, as there is only one unreachable state.
+                if (!this.Reachable) return;
+                index *= 2;
+                _state[index] = (value is NullableFlowState.NotNull or NullableFlowState.MaybeNull);
```

Comment:
It feels like it would be a little more clear to assign these using a switch expression which is similar in shape to the one in `GetValue(int)` above.